xontrib/tests/test_uncpushd.py

```diff
@@ -0,0 +1,134 @@
+'''Tests for xontrib.uncpushd'''
+
+import pytest
+import os
+from xontrib.uncpushd import unc_pushd, unc_popd, _unc_tempDrives, _do_subproc
+
+import builtins
+from xonsh.environ import Env
+from xonsh.built_ins import load_builtins
+from xonsh.dirstack import DIRSTACK
+from xonsh.platform import ON_WINDOWS
+
+
+@pytest.fixture(scope="module")
+def wd_setup(request):
+    """
+    test fixture creates two subdirectories under test folder containing this test file, invokes each test
+    with working directory set to tests folder and DIRPATH and PWD set in builtins.__xonsh_env__
+    (as is apparently needed by xonsh/dirstack.py, revealed by a close reading of their unit tests.)
+    """
+
+    _old_pwd = os.getcwd()
+    _start_wd = os.path.dirname(os.path.abspath(__file__))
+
+    try:
+        os.mkdir( os.path.join( _start_wd, 'cdtest1'))
+    except FileExistsError:
+        pass
+    try:
+        os.mkdir( os.path.join( _start_wd, 'cdtest2'))
+    except FileExistsError:
+        pass
+
+    load_builtins()
+    _old_env = builtins.__xonsh_env__
+    builtins.__xonsh_env__ = Env(CDPATH=os.path.dirname(_start_wd), PWD=_start_wd)
+    os.chdir( _start_wd)
+
+    def fin():
+        builtins.__xonsh_env__ = _old_env
+        os.chdir(_old_pwd)
+
+    request.addfinalizer( fin)
+
+    return _start_wd
```

Comment:
I don't think the temporary directories should be created in the tests directory. Pytest has the notion of a basetmp directory associated with a test run. You easily create temp directories with the `tmpdir_factory` fixture. See http://pytest.org/latest/tmpdir.html#the-tmpdir-factory-fixture

You can also simplify this a lot with the pytest yield fixtures. Then you don't need the `request.addfinalizer( fin)`

```python
@pytest.fixture(scope="module")
 def wd_setup(tmpdir_factory):
     #setup code.
     ...
     yield _start_wd
     # finalizer code
     ...
```